driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -153,21 +152,21 @@ private static void validateArrayFilters(final ConnectionDescription connectionD
         }
     }
 
-    private static void validateHint(final ConnectionDescription connectionDescription, final WriteConcern writeConcern,
-                                     final WriteRequest request) {
+    private static void validateWriteRequestHint(final ConnectionDescription connectionDescription, final WriteConcern writeConcern) {
```

Comment:
These two methods appear to be exactly the same.  Any reason not to just delete this one and use `validateHint` for all callers?

Reply:
I went too far with this change. The latest update to this method conforms to the spec.

Reply:
I don't think you did.  It seems like the new condition you added

    (request instanceof DeleteRequest || request instanceof UpdateRequest)

will always be true, so the condition can be removed, and with that the `request` parameter.  And then the two methods become identical again.

Reply:
There is a slight difference between the two methods. For write requests, the first condition to check is if the server is less than 3.4. But for `find` delete and update operations, the first condition to check is if the server is less than 4.2.

Reply:
Got it.